Fix for issue #401 - Unable to create a pod with "resources: limits" such as "nvidia.com/gpu: 1"

simonbrommers opened the pull request:
My first attempt at a more serious PR for an OS project - please go easy on me..
It's only small in nature - but it fixes a big problem for my use-case - needing to target Spring Cloud Data Flow tasks to GPU capable nodes.
I'm unsure if I'm making this PR to the correct target branch - please advise if not.  My aim is to get the code to a release ASAP.

simonbrommers commented:
Thanks for your patience and pointers @chrisjs - I've changed the target branch for the PR to master now, and added tests for the changes I made.  I think the formatting should be ok.. I know I initially made a mess of it, but I think I cleared it up.

chrisjs commented on src/main/java/org/springframework/cloud/deployer/spi/kubernetes/KubernetesDeployerProperties.java:
```diff
@@ -81,12 +81,24 @@ public void setFabric8(Config fabric8) {
 		private String memory;
 
+		/**
+		 * Container GPU vendor limit.
```
"Container GPU vendor name for limit" - this content is extracted and auto populated into UI components, and seems a little more descriptive

chrisjs reviewed (changes requested):
this looks good, thanks. i made a couple comments on minor things. once those are patched up ill go ahead and merge it.

simonbrommers commented:
I've addressed the comments you made now - fingers crossed.
With the hope this will now be accepted and merged, what should I expect in terms of a new release?  Is there something that documents the process that takes place?  I got the feeling milestone builds are in the mix somewhere before an actual release

chrisjs commented:
squashed and merged as 1e8d60cad804d9afe0531c54bee04e7516ba90ef 

release order is SNAPSHOT -> Milestone(s) -> Release Canadiate(s) -> RELEASE. this change will go into the upcoming RELEASE, @sabbyanandan would be able to provide a better estimate on when that will occur

thanks again for the contrib!
